Move the taproot module to crate root

tcharding opened the pull request:
We are trying to flatten the `util` module. The `taproot` module can live in the crate root. If/when we create a `crypto` module/crate we may wish to pull some stuff out of this module but for now moving it gets us closer to removing `util` without making the directory structure any worse.

CC sanket1729, I did this after reviewing the `taproot` module during work on https://github.com/rust-bitcoin/rust-bitcoin/pull/1260 and your review comment.

apoelstra reviewed (dismissed):
ACK bbad1af96248a55c7a88fc2a5e59b5ed211fb4ed

Kixunil reviewed (dismissed):
ACK bbad1af96248a55c7a88fc2a5e59b5ed211fb4ed

tcharding commented:
\me whispers excitedly merge merge merge merge merge

tcharding commented:
Rebased! Felt so good to see `psbt` in `bitcoin/src/` for some reason :)

apoelstra reviewed (dismissed):
ACK 676f9c2bb5cf8d6ac5c2d9651ec4543de9495a19

tcharding commented:
Rebased, no other changes.

apoelstra reviewed (dismissed):
ACK 43091087e214c0e5c8827ff0892a8d01e799aee9

tcharding commented:
No changes since you last ack'ed @Kixunil, here is the range diff command for you to verify what I say, can I get another ack please `git range-diff bbad1af...@`

Kixunil reviewed (approved):
ACK 43091087e214c0e5c8827ff0892a8d01e799aee9

tcharding commented:
Rebased and added `rustfmt::skip` attributes to the `sha256t_hash_newtype!` macro calls. Added an additional patch that does the formatting. I hope I haven't just made re-acking this painful ...

Kixunil commented on bitcoin/src/taproot.rs:
```diff
@@ -481,20 +459,17 @@ impl TaprootBuilder {
                     unreachable!("Size checked above. Builder guarantees the last element is Some")
-                }
-            }
+                },
```
I may be paranoid but this style seems to ask for heartbleed-style vulnerability.

apoelstra replied:
Heh, I agree it feels a bit treacherous, but Rust will at least require you put **something** (not just whitespace) before any following lines of code, so you can't be easily fooled into mistaking which block it belongs to.

Kixunil reviewed (approved):
ACK dce88b09ff65a190f617b1f1487c50b04e73e1ce

apoelstra reviewed (approved):
ACK dce88b09ff65a190f617b1f1487c50b04e73e1ce
